src/cli/flags/repo.go

```diff
@@ -33,6 +33,16 @@ func AddBackupIDFlag(cmd *cobra.Command, require bool) {
 	}
 }
 
+// ---------------------------------------------------------------------------
+// storage
+// ---------------------------------------------------------------------------
+
+func AddAllStorageFlags(cmd *cobra.Command) {
+	AddCorsoPassphaseFlags(cmd)
+	AddS3BucketFlags(cmd)
```

Comment:
S3 bucket and AWS creds flags are valid for AWS S3 storage only. Calling this function from `corso backup * *` or `corso restore * *` will expose these flags for other storage providers. Additionally, this won't expose any flags for other storage providers like filesystem. We should fix this. 

Also, quick sanity check: We are adding `AddS3BucketFlags` here. We currently only expose these flags for `repo * * `, and config file serves as the source of truth for bucket params for other commands. Given we `matchFromConfig`, any discrepancies between flags & config file should lead to an error, but we should validate this.

Reply:
> S3 bucket and AWS creds flags are valid for AWS S3 storage only. 

Correct, but every operation command needs to expose the total set of storage provider flags, because those commands are not scoped to a specific provider.  IE: it's `corso restore exchange` not `corso restore s3 exchange`.

> Additionally, this won't expose any flags which are meant for providers like filesystem

The main branch at the time of this PR didn't contain those flags.  I'll update the code in future PRs as needed.

> We are adding AddS3BucketFlags here. We currently only expose these flags for repo * * ,

Ah, good catch, will fix.

Reply:
@pandeyabs `AddS3BucketFlags()` has been removed from the func.  Can you give this PR another review, please?

Also, since you jumped in, would you kindly also review the parent PR?  https://github.com/alcionai/corso/pull/4294  Thanks!

Reply:
> Not saying we shouldn't expose bucket flags. It shouldn't break anything IMO. Just asking to verify functionality

I think it's better to keep flags out unelss the command explicitly supports them.  If we don't want the user overriding the s3 bucket during these commands, we shouldn't expose them.  Likewise, no-op flags will confuse end users, and should also be avoided.